disassemblers/ofrak_ghidra/ofrak_ghidra/components/ghidra_analyzer.py

```diff
@@ -220,6 +225,51 @@ def _build_ghidra_server_args(self) -> List[str]:
 
         for ghidra_script in self._scripts:
             args.append(ghidra_script)
-            args.append(ghidra_script.split(".")[0].lower())
 
         return args
+
+
+@dataclass
+class GhidraCodeRegionConfig(ComponentConfig):
+    pass
+
+
+class GhidraCodeRegionModifier(Modifier[Optional[GhidraCodeRegionConfig]], OfrakGhidraMixin):
+    id = b"GhidraCodeRegionModifier"
+    targets = (CodeRegion,)
+
+    get_code_regions_script = OfrakGhidraScript(
+        os.path.join(CORE_OFRAK_GHIDRA_SCRIPTS, "GetCodeRegions.java"),
+    )
+
+    async def modify(self, resource: Resource, config: Optional[GhidraCodeRegionConfig] = None):
+        code_region = await resource.view_as(CodeRegion)
+        ghidra_project = await OfrakGhidraMixin.get_ghidra_project(resource)
+
+        ofrak_code_regions = await ghidra_project.resource.get_descendants_as_view(
+            v_type=CodeRegion, r_filter=ResourceFilter(tags=[CodeRegion])
+        )
+
+        backend_code_regions_json = await self.get_code_regions_script.call_script(resource)
+        backend_code_regions = []
+
+        for cr_j in backend_code_regions_json:
+            cr = CodeRegion(cr_j["start"], cr_j["size"])
+            backend_code_regions.append(cr)
+
+        ofrak_code_regions = sorted(ofrak_code_regions, key=lambda cr: cr.virtual_address)
+        backend_code_regions = sorted(backend_code_regions, key=lambda cr: cr.virtual_address)
+
+        for cr in ofrak_code_regions:
+            if cr.virtual_address == code_region.virtual_address:
+                relative_va = cr.virtual_address - ofrak_code_regions[0].virtual_address
+
+                for backend_cr in backend_code_regions:
+                    backend_relative_va = (
+                        backend_cr.virtual_address - backend_code_regions[0].virtual_address
+                    )
+
+                    if backend_relative_va == relative_va and backend_cr.size == cr.size:
+                        cr.resource.add_view(backend_cr)
+
+        LOGGER.debug(f"Could not find CodeRegion for {code_region}")
```

Comment:
Looks like this will be printed every time the modifier runs, could also be clearer in what it's saying:
```suggestion
                        cr.resource.add_view(backend_cr)
                        return

        LOGGER.debug(f"Could not find matching Ghidra CodeRegion for {code_region}")
```

Reply:
This will only print if there was no code region found in ghidra that matches the relative offset and size of the one in ofrak

Reply:
In the current version of this PR, there is no early exit in that case , so the log message is always printed. That's why in my suggestion I've added a `return` statement.

Reply:
oh whoops